docs/how-to-guides/develop-a-tool/index.md

```diff
@@ -12,4 +12,5 @@ use-file-path-as-tool-input
 customize_an_llm_tool
 create-cascading-tool-inputs
 create-your-own-custom-strong-type-connection
+create-dynamic-list-tool-input
```

Comment:
If need to add doc to https://github.com/microsoft/promptflow/blob/main/docs/how-to-guides/develop-a-tool/create-and-use-tool-package.md#advanced-features as well.